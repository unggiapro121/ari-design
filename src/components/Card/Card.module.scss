@import '../../styles/variables';
@import '../../styles/mixins';

.card {
  margin: 10px;
  background-color: $color-white;
  padding: $spacing-l;
  border: 1px solid $color-light-grey;
}
