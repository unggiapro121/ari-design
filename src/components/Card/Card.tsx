import React, { CSSProperties } from 'react';

import styles from './Card.module.scss';

export interface CardProps {
  children?: React.ReactNode;
  style?: CSSProperties;
  className?: string;
}

/**
 * Basic Card component
 * @param children
 * @param style
 * @param className
 */
const Card = ({ children, style, className }: CardProps) => {
  return (
    <div className={`${styles['card']} ${className}`} style={style} data-testid="Card">
      {children}
    </div>
  );
};

export default Card;
