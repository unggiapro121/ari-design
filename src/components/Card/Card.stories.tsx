import { Meta, Story } from '@storybook/react';
import * as React from 'react';

import Card from './Card';

export default {
  title: 'In Progress/Cards',
  component: Card,
  parameters: {
    backgrounds: {
      default: 'dark',
    },
    zeplinLink: [
      {
        name: 'Wireframe',
        link: '...',
      },
    ],
  },
} as Meta<typeof Card>;

const Template: Story<typeof Card> = (args) => <Card {...args}>
  <h1>Title</h1>
  <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,
    molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum
    numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium
    optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis
    obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam
    nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit,
    tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit,
    quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos
    sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam
    recusandae alias error harum maxime adipisci amet laborum. Perspiciatis
    minima nesciunt dolorem! Officiis iure rerum voluptates a cumque velit
    quibusdam sed amet tempora. Sit laborum ab, eius fugit doloribus tenetur
    fugiat, temporibus enim commodi iusto libero magni deleniti quod quam
    consequuntur! Commodi minima excepturi repudiandae velit hic maxime
    doloremque. Quaerat provident commodi consectetur veniam similique ad
    earum omnis ipsum saepe, voluptas, hic voluptates pariatur est explicabo
    fugiat, dolorum eligendi quam cupiditate excepturi mollitia maiores labore </p>
</Card>;

export const Primary = Template.bind({});
Primary.args = { };