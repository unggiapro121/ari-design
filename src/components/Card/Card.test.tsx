import React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card';

describe('Card component', () => {
  test('does render correctly', () => {
    const component = renderer.create(<Card><h1> I'm a card! </h1></Card>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
