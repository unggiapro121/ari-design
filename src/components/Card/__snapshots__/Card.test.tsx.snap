// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > does render correctly 1`] = `
<div
  className="_card_d75732 undefined"
  data-testid="Card"
>
  <h1>
     I'm a card! 
  </h1>
</div>
`;
